### PR TITLE
Rename downloadSecurity to downloadStrategy

### PR DESCRIPTION
**Pull request by core23**

### CODE REVIEW

File: Provider/Pool.php [hunk 1]
```diff
@@ -68,12 +76,24 @@ public function addProvider($name, MediaProviderInterface $instance)
...
     /**
-     * @param string                                                 $name
-     * @param \Sonata\MediaBundle\Security\DownloadStrategyInterface $security
+     * @deprecated since version 2.4, to be removed in 3.0.
      */
     public function addDownloadSecurity($name, DownloadStrategyInterface $security)
     {
+        @trigger_error('The '.__METHOD__.' method is deprecated since version 2.4 and will be removed in 3.0.', E_USER_DEPRECATED);
+        
         $this->downloadSecurities[$name] = $security;
```
Comment (Soullivaneuh) [inline Provider/Pool.php]: Should be on `addDownloadStrategy` method.

File: Provider/Pool.php [hunk 2]
```diff
@@ -213,23 +233,33 @@ public function getProviderList()
...
 
     /**
-     * @param \Sonata\MediaBundle\Model\MediaInterface $media
+     * @deprecated since version 2.4, to be removed in 3.0.
+     */
+    public function getDownloadSecurity(MediaInterface $media)
+    {
+        @trigger_error('The '.__METHOD__.' method is deprecated since version 2.4 and will be removed in 3.0.', E_USER_DEPRECATED);
+        
+        return array_merge($this->getDownloadSecurity($media), $this->getDownloadStrategy($media));
```
Comment (Soullivaneuh) [inline Provider/Pool.php]: No array_merge, just call `getDownloadStrategy`. See https://github.com/sonata-project/SonataMediaBundle/pull/872/files#r43988726.

Comment (core23) [inline Provider/Pool.php]: IMO Calling ``array_merge`` is more BC, if you add something directly to the ``$this->downloadSecurities`` without calling any method.

File: Provider/Pool.php (hunk 1 above)
Comment (core23) [inline Provider/Pool.php]: Are you sure? The ``addDownloadStrategy`` method is new.

File: Provider/Pool.php (hunk 2 above)
Comment (Soullivaneuh) [inline Provider/Pool.php]: You mean for class extension?

**Comment by Soullivaneuh** [comment]
This PR need some tests.

File: Provider/Pool.php (hunk 2 above)
Comment (core23) [inline Provider/Pool.php]: Correct. If you extend this class you could have access the ``downloadSecurities`` as they are protected.

Comment (Soullivaneuh) [inline Provider/Pool.php]: Ok then, please just add some tests to ensure this is working.

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Comment by OskarStark** [comment]
Please resubmit this against 3.x and create a PR for 4.0 afterwards

**Pull request closed without merging**